bundle/src/main/java/com/adobe/acs/commons/adobeio/core/service/EndpointService.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2013 - 2018 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.adobeio.core.service;
+
+import java.util.Map;
+
+import com.adobe.acs.commons.adobeio.core.types.Filter;
+import com.adobe.acs.commons.adobeio.core.types.PKey;
+import com.google.gson.JsonObject;
+
+/**
+ * This is the interface for the EndPointService
+ * Using this service, the calling component can post to and retrieve from Service.
+ *
+ * Example how to use in the calling component:
+ *   @Reference(target = "(getId=///put your id here///)")
+ *   private EndpointService endpointService;
```

Comment:
I'm confused by this -- it looks like the purpose of the EndpointServiceFactory is to get an EndpointService by id, but here you have it being injected directly. Why two patterns to do the same thing?

Reply:
Is this `@ProviderType`? If so, annotation should be added